tests/message/unittest_message_id_store.py

```diff
@@ -90,3 +94,47 @@ def test_duplicate_msgid(msgid_store: MessageIdStore) -> None:
         "Message symbol 'warning-symbol' cannot be used for 'W1234' and 'W1235'"
         in str(error.value)
     )
+
+
+def test_exclusivity_of_msg_ids() -> None:
+    """Test to see if all checkers have an exclusive message id prefix"""
+    err_msg = (
+        "{} has the same prefix ('{}') as the '{}' checker. Please make sure the prefix "
+        + "is unique for each checker. You can use 'script/get_unused_message_id_category.py' "
+        + "to get an unique id."
+    )
+
+    runner = lint.Run(
+        ["--enable-all-extensions", join(HERE, "..", "regrtest_data", "empty.py")],
+        exit=False,
+    )
+
+    # Some pairs are hard-coded as they are pre-existing and non-exclusive
+    # and we (currently) don't want to rename them
```

Comment:
```suggestion
    # Some pairs are hard-coded as they are pre-existing and non-exclusive
    # and we don't want to rename them and make old msgid doc obsoletes
```
I think this will never be worth it. It would make sooooo much old names and break every question asking about a specific msgid on stackoverflow.